### PR TITLE
feat:  Add web component core package

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

### CODE REVIEW

File: packages/core/src/css.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export function css(templateData: TemplateStringsArray, ...args: any[]): string {
+    let retVal = templateData[0];
+    for (let i = 0; i < args.length; ++i) {
+        retVal += String(args[i]);
+        retVal += templateData[i + 1];
```
Comment (jeclrsg) [inline packages/core/src/css.ts:5]: I've not done much with template strings like this, so I don't really follow how this is used. Could this not be out of bounds / undefined?

File: packages/core/src/message.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Message } from "@hpcc-js/util";
+import { HPCCElement } from ".";
+
+export interface Change<T> {
+    oldValue: T;
+    newValue: T;
+}
+
+export type ChangeMap<T extends HPCCElement> = {
+    [what in keyof T]+?: Change<T[what]>;
```
Comment (jeclrsg) [inline packages/core/src/message.ts:10]: What is this [...]+?: ...? I wasn't even sure what to search for to try and figure that out 😄

File: packages/core/src/html.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            directives.push(arg);
+            if (arg instanceof Ref) {
+                html += `id="${arg.propertyName}"`;
+            } else {
+                html += String(values[i]);
+            }
+        } else {
+            throw new Error("Invalid Directive");
+        }
+        html += strings[i + 1];
```
Comment (jeclrsg) [inline packages/core/src/html.ts:33]: Likewise here, as mentioned in the css.ts

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith no changes, I don't think. Just a couple or three questions. Unless those array index spots should be checked.

File: packages/core/src/css.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/core/src/css.ts:5]: templateData is an array which is always 1 larger than the args array per the specification.

File: packages/core/src/html.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/core/src/html.ts:33]: Again this is a normal pattern as the "strings" will always be an array 1 larger than the values.

File: packages/core/src/message.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/core/src/message.ts:10]: See "Mapping Modifiers":  https://www.typescriptlang.org/docs/handbook/2/mapped-types.html

File: packages/core/src/svg.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    enter() {
+    }
+
+    update(changes: ChangeMap<this>) {
+        super.update(changes);
+        if ("resize" in changes) {
+            this._svg.setAttribute("width", `${this.clientWidth}px`);
+            this._svg.setAttribute("height", `${this.clientHeight - 5}px`);
```
Comment (jbrundage) [inline packages/core/src/svg.ts:22]: What is this -5 for?

**Pull request merged**